rust_src/src/base64.rs

```diff
@@ -157,16 +257,12 @@ fn base64_encode_string(string: LispObject, no_line_break: LispObject) -> LispOb
     let encoded = buffer.as_mut_ptr();
     let encoded_length = base64_encode_1(
         string.sdata_ptr(),
-        encoded,
         length,
+        encoded,
+        allength,
         no_line_break.is_nil(),
         string.is_multibyte(),
     );
-
-    if encoded_length > allength {
```

Comment:
Is this check no longer worthwhile?

Reply:
Correct. The base64 code already checked it.

Reply:
I hope to update the upstream crate to export the computation to make our allocation better.

Reply:
Currently the upstream code panics if the size is off. I also hope to get a Result returning API accepted.

Reply:
Sounds good to me. 👍